src/API/Common/Event/Dispatcher.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Common\Event;
+
+use CloudEvents\V1\CloudEventInterface;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextKey;
+
+class Dispatcher
+{
+    private static ?ContextKey $key = null;
+    /** @var array<string, array<int, array<callable>>> */
+    private array $listeners = [];
+
+    private function __construct()
+    {
+    }
+
+    public static function getInstance(): self
+    {
+        self::$key ??= new ContextKey(self::class);
+        $dispatcher = Context::getCurrent()->get(self::$key);
+        if ($dispatcher === null) {
+            $dispatcher = new self();
+            Context::getCurrent()->with(self::$key, $dispatcher)->activate();
```

Comment:
Could you make this a separate method (`createInstance`)?

Reply:
si. done.